core/server/master/src/main/java/alluxio/master/meta/checkconf/ServerConfigurationRecord.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta.checkconf;
+
+import alluxio.PropertyKey;
+import alluxio.wire.Address;
+import alluxio.wire.ConfigProperty;
+
+import com.google.common.base.Preconditions;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.stream.Collectors;
+
+/**
+ * This class is responsible for recording server-side configuration.
+ */
+public class ServerConfigurationRecord {
```

Comment:
- is the class supposed to be threadsafe? if yes, can you annotate it? 
btw, do we need it to be thread-safe here?
- I think calling this `ConfigurationRecord` is confusing because you also have `ConfigRecord` which is a single record of configuration, whereas this is the entire collection of configuration state across multiple servers. Maybe `{Server}ConfigurationStore` or maybe just `{Server}ConfigurationCollection` -- I am not sure how important to have `server` in the name, do we filter out all client-side conf?

Reply:
Yes, we only check the Property keys which scope includes MASTER or WORKER, so client-only configuration will be filtered out. ServerConfigurationStore is good.

Reply:
Added thread safe annotation, thanks